### ISSUE TITLE
False positive in require-yield where yield is anywhere but function scope

### ISSUE THREAD

**Issue by maurice** [oxc-project/oxc#2323]
*Working*

```js
function* foo() { yield 0; }
```

*Not working (simplified test case)*

```js
function* foo() { while (true) { yield 0; } }
```

*False positive result*

```
  ⚠ eslint(require-yield): This generator function does not have 'yield'
   ╭─[require_yield.tsx:1:11]
 1 │ function* foo() { while (true) { yield 0; } }
   ·           ───
   ╰────
```

**Comment by Boshen** [comment]
This is regression from #2208, where @Dunqing removed the function stack from the semantic builder. I'll get it reverted and your test case 😅
